server/Services/PosterService.cs

```diff
@@ -83,5 +92,27 @@ public Poster UpdatePoster(int id, Poster poster)
         }
 
         public Object UpdatePosterJSON(int id, Poster p) => UpdatePoster(id, p).ToJSON();
+        public string SanityCheck(Poster p)
+        {
+            if (p.Name == null || p.Name == "")
+            {
+                return "Name cannot be empty";
+            }
+
+            if (string.IsNullOrEmpty(p.ImageUrl) || string.IsNullOrWhiteSpace(p.ImageUrl))
+            {
+                return "Image url cannot be empty";
+            }
+            if (p.StartDate == null)
+            {
+                return "Start date cannot be empty";
+            }
+            if (p.EndDate == null)
+            {
+                return "End date cannot be empty";
+            }
+
+            return "";
+        }
```

Comment:
This seems like an awkward way to do this. Could we instead handle this with exceptions in the getters and setters of the Poster class.